metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DatasetService.java

```diff
@@ -126,6 +127,17 @@ public Dataset createDataset(MetisUser metisUser, Dataset dataset)
             .format("Dataset with organizationId: %s and datasetName: %s already exists..",
                 dataset.getOrganizationId(), dataset.getDatasetName()));
       }
+
+      // Verify references to old datasetIds
+      if (dataset.getOldDatasetIds() != null) {
+        for (String id : dataset.getOldDatasetIds()) {
+          if (datasetDao.getDatasetByDatasetId(id) == null) {
+            throw new BadContentException(
+                String.format("Old datasetId %s doesn't exist", id));
+          }
+        }
+      }
```

Comment:
I don't think we can assume that the IDs necessarily exist? What if the old dataset has been removed (manually as part of cleanup)? But we still want the redirects to exist? Is this a use case we should consider? Or should we move to a system where we only invalidate datasets (and not remove them) so that IDs are not reused?

Reply:
I'm actually not sure on how to answer that.
I assumed that if we did a cleanup of a dataset we should remove it's reference from other datasets.
The validation is here to verify that a user will not add a reference to a dataset that doesn't exist.
Maybe we should analyze this a bit

Reply:
For this I'm removing the references when a dataset gets deleted.